relay_messages_test.go

```diff
@@ -307,6 +310,68 @@ func TestLazyCallReqSetTTL(t *testing.T) {
 	})
 }
 
+func byteKeyValToMap(tb testing.TB, buffer []byte) map[string]string {
+	rbuf := typed.NewReadBuffer(buffer)
+	nh := int(rbuf.ReadSingleByte())
+	retMap := make(map[string]string, nh)
+	for i := 0; i < nh; i++ {
+		keyLen := int(rbuf.ReadSingleByte())
+		key := rbuf.ReadBytes(keyLen)
+		valLen := int(rbuf.ReadSingleByte())
+		val := rbuf.ReadBytes(valLen)
+		retMap[string(key)] = string(val)
+	}
+	require.NoError(tb, rbuf.Err())
+	return retMap
+}
+
+func uint16KeyValToMap(tb testing.TB, buffer []byte) map[string]string {
```

Comment:
I think this is the same as the `decodeThriftHeaders` function that's in `relay_test.go`? We can move it to a test package to make it easier, but shouldn't need to reimplement it

Reply:
turns out there's any import cycle preventing that from being reused: `tchannel` imports `thrift` (`thrift.ReadHeaders()`), which imports `tchannel`. added TODO to change this once we have bandwidth to do a large-ish refactor to resolve this.